codes/javascript/chapter_stack_and_queue/linkedlist_deque.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * File: linkedlist_deque.js
+ * Created Time: 2023-02-04
+ * Author: Zhuo Qinyue (1403450829@qq.com)
+ */
+
+class DequeNode {
+    prev;   // 前驱结点引用 (指针)
+    next;   // 后继结点引用 (指针)
+    val;    // 结点值
+    
+    constructor(val) {
+        this.val = val;
+        this.next = null;
+        this.prev = null;
+    }
+}
+
+/* 基于双向链表实现的双向队列 */
+class LinkedListDeque {
+    head;   // 头结点 head
+    tail;   // 尾结点 tail
+    len;    // 双向队列的长度
```

Comment:
size

Reply:
Properties and methods in the js class cannot have the same name, so I use `len`.
If I name the property `size`, then I will name the method `getSize()`